src/server/acl/validator.cc

```diff
@@ -4,12 +4,23 @@
 
 #include "server/acl/validator.h"
 
+#include "base/logging.h"
+#include "server/acl/acl_commands_def.h"
+#include "server/server_state.h"
+
 namespace dfly::acl {
 
 [[nodiscard]] bool IsUserAllowedToInvokeCommand(const ConnectionContext& cntx,
                                                 const facade::CommandId& id) {
-  auto command_credentials = id.acl_categories();
-  return (cntx.acl_categories & command_credentials) != 0;
+  auto cat_credentials = id.acl_categories();
+
+  const auto& commands = acl::CommandsIndexer();
+  DCHECK(commands.contains(id.name()));
+  auto [index, command_mask] = commands.find(id.name())->second;
+  DCHECK_LT(index, cntx.acl_commands.size());
```

Comment:
Also small nit, maybe you can store the command indexer result in main_service or somewhere else instaead of global. Like adjacent to where the commandregistry is stored now

IIRC  IsUserAllowedToInvokeCommand( is called from VerifyCommand that should have access to the service

Reply:
Need to think how to get rid of the num families global variable 🤔

Reply:
Honestly that global variable is not that bad, and it's initialization is protected. Maybe there is a better pattern? :shrug:

Reply:
Sure, my only question is how? This should be initialized once and it should be `const`, how can you enforce `constness` of a member variable that is initialized after construction?

Reply:
You can enforce constness via a wrapper but that's an overkill (this is what I did originally tbh)